topics/icrc_29_window_post_message_transport.md

```diff
@@ -43,23 +44,32 @@ The signer should reply with a message indicating that it is ready to receive ad
 
 > **Note**: The relying party should send `icrc29_status` messages in short intervals. It is expected that some of the messages will be lost due to being sent before the signer is ready.
 
-After the `"result": "ready"` response has been received, the relying party can send [ICRC-25](https://github.com/dfinity/wg-identity-authentication/blob/main/topics/icrc_25_signer_interaction_standard.md) messages to the signer
+After the `"result": "ready"` response has been received within a reasonable timeframe, the relying party can send [ICRC-25](https://github.com/dfinity/wg-identity-authentication/blob/main/topics/icrc_25_signer_interaction_standard.md) messages to the signer
 using the [window.postMessage](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) API.
 
 ## Authentication
 
-* The relying party must authenticate the signer by the `origin` of the window that it opens.
-* The signer must authenticate the relying party by the `origin` property of the `message` event that it received.
+* The relying party must authenticate the signer by checking if both:
+  - The received message `source` property is equal to the `window` that was opened.
+  - The received message `origin` property is equal to the `origin` property in the `"result": "ready"` response received when communication channel was established.
+* The signer must authenticate the relying party by checking if both:
+  - The received message `source` property is equal to the `source` property in the `icrc29_status` message received when communication channel was established.
+  - The received message `origin` property is equal to the `origin` property in the `icrc29_status` message received when communication channel was established.
 
 ## Sending Messages
 
-Messages are sent by calling `window.postMessage` on the signer window, or the `window.opener` respectively.
-When sending messages, the `targetOrigin` parameter must be set to the origin of the signer or relying party window.
+Messages are sent by calling `window.postMessage` on the signer `window`, or the `source` property in the `icrc29_status` message received when communication channel was established respectively.
```

Comment:
How about having two subsections within `Sending Messages`. One for the relying party and another for the signer.

Or make it two sections: `Sending Messages: Signer` and `Sending Messages: Reying Party`.

The recommendations here are intertwined and a dev building one or the other should have it easy to find how to deal with this. The same way that the `Authentication` section has two points.

Reply:
Good points, it was indeed getting harder to read 😅